experimental/rank1_bnns/refining.py

```diff
@@ -0,0 +1,119 @@
+# coding=utf-8
+# Copyright 2020 The Edward2 Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Lint as: python3
+"""Utilities for sampling"""
+import numpy as np
+import tensorflow.compat.v2 as tf
+import tensorflow_probability as tfp
+
+from experimental.rank1_bnns.rank1_bnn_layers import DenseRank1, Conv2DRank1
+
+
+def get_auxiliary_posterior(posterior_mean,
+                            posterior_scale,
+                            prior_mean,
+                            prior_scale,
+                            auxiliary_scale):
+  """ Calculate the posterior distribution of an additive Gaussian """
+  """ auxiliary variable. q(a)=\int p(a|w)q(w)dw. """
+  prior_var = tf.math.pow(prior_scale, 2)
+  posterior_var = tf.math.pow(posterior_scale, 2)
+  auxiliary_var = tf.math.pow(auxiliary_scale, 2)
+  auxiliary_posterior_mean = (posterior_mean - prior_mean) \
+      * auxiliary_var / prior_var
+  auxiliary_posterior_var = posterior_var * tf.math.pow(auxiliary_var, 2) \
+      / tf.math.pow(prior_var, 2) + auxiliary_var * \
+      (prior_var - auxiliary_var) / prior_var
+  return auxiliary_posterior_mean, tf.sqrt(auxiliary_posterior_var)
+
+
+def get_conditional_prior(prior_mean,
+                          prior_scale,
+                          auxiliary_scale,
+                          auxiliary_sample):
+  """ Calculate the conditional prior given the value of an additive """
+  """ Gaussian auxiliary variable. p(w|a). """
+  prior_var = tf.math.pow(prior_scale, 2)
+  auxiliary_var = tf.math.pow(auxiliary_scale, 2)
+  return prior_mean + auxiliary_sample, tf.sqrt(prior_var - auxiliary_var)
+
+
+def get_conditional_posterior(posterior_mean,
+                              posterior_scale,
+                              prior_mean,
+                              prior_scale,
+                              auxiliary_scale,
+                              auxiliary_sample):
+  """ Calculate the conditional posterior given the value of an additive """
```

Comment:
Use

```python
"""Calculate the conditional posterior given the value of an additive 
Gaussian auxiliary variable. q(w|a)\propto p(a|w)q(w).
"""
```

That is, no space between triple quotes, and one only triple quote block (the second one here is actually a string not encoded in the function's `__doc__` attribute).

In general, note also the recommended PEP-8 style is to use descriptive instead of imperative actions (calculates instead of calculate) and to have the first sentence in the docstring fit in the first line.

Reply:
+1

Reply:
Changed all the docstrings